src/Shared/E2ETesting/WaitAssert.cs

```diff
@@ -50,10 +50,13 @@ public static void Exists(this IWebDriver driver, By finder)
         public static void Exists(this IWebDriver driver, By finder, TimeSpan timeout)
             => WaitAssertCore(driver, () => Assert.NotEmpty(driver.FindElements(finder)), timeout);
 
-        public static IWebElement WaitUntilExists(this IWebDriver driver, By findBy) =>
-            WaitUntilExists(driver, findBy, default);
+        public static IWebElement Element(this IWebDriver driver, By findBy)
+        {
+            driver.Exists(findBy);
+            return driver.FindElements(findBy)[0];
```

Comment:
The naming of these `IWebElement` extension methods bothers me a bit, because nothing in their names makes clear which of them tolerate waiting until the thing exists or is true, and which of them will return null or fail of the thing/condition isn't initially true.

It's not enough just to say "we should always wait" because (1) the native Selenium methods don't, and nothing stops us using them, and (2) it's not always desirable to wait. For the tests to be as precise as possible, in scenarios where we know something should already be true, it's better to use a lookup/assertion that would fail immediately if it couldn't be satisfied on the first attempt.

How would you feel about changing our "wait until ... " extension methods to have names that all start with "Wait"? I have thought this for a while about methods like `Browser.AssertX`, which I always thought should be called `Browser.WaitAssertX` for clarity.

If you would be happy with this change but don't want to do it in this PR, that's totally fine with me. If we have agreement we can create a follow-up issue for it, which I don't mind doing if you're busy with other things.

Reply:
Should we simply integrate all of them within Exists and have Exists return the element if found? I was trying to avoid making this PR into an "assertion consolidation PR", but I'm fine taking that change so long we are all ok with it.

To be clear the proposal would be something like

Exists `IWebElement Exists(this IWebDriver driver, Func<IWebElement> elementSelector, TimeSpan timeout = default)

Reply:
+1 to integrating them into a smaller number of methods.

My question about the naming still stands though. Do you have any objection to using the term `Wait` in the method names for all the things that implicitly wait?

Reply:
* I've integrated everything under Browser.Exist (which now returns the found element).
* Every method in `WaitAssert.cs` implicitly waits for the assertion to be true or fails otherwise.
* This is inline with how `Assert.Single` and similar methods work on xUnit.